app/controllers/application_controller.rb

```diff
@@ -18,8 +20,64 @@ def set_checkout_redirect
     end
   end
 
+  def enable_embedded_styles
+    session[:embedded_shopfront] = true
+    render json: {}, status: 200
```

Comment:
This is weird as well as I  can call `$.ajax({url: '/embedded_shopfront/enable',type: 'POST'});` from browser console without any problem to enable session. There are two ways where we can enable it seems vulnerable to hack as well.